module/zfs/zfs_ioctl.c

```diff
@@ -2984,9 +2986,13 @@ zfs_ioc_set_prop(zfs_cmd_t *zc)
 	 */
 	curtargetip[0] = '\0';
 	(void) nvlist_lookup_string(nvl, ZFS_PROP_TARGET_IP, &targetip);
-	if (targetip != NULL)
+	if (targetip != NULL) {
 		error = zfs_set_targetip_prehook(zc->zc_name, source, targetip,
 		    &curtargetip[0]);
+	} else if (error == 0 && nvlist_lookup_string(nvl,
```

Comment:
lets allow setting  both (targetip and readonly) at once if we want to?

Reply:
we can do that, but that will complicate creation part i think. We can plan this in next release.

Reply:
what is the problem if we remove 'else' which allows both the settings?